logging/zap/grpclogger.go

```diff
@@ -44,3 +47,103 @@ func (l *zapGrpcLogger) Printf(format string, args ...interface{}) {
 func (l *zapGrpcLogger) Println(args ...interface{}) {
 	l.logger.Info(fmt.Sprint(args...))
 }
+
+// ReplaceGrpcLoggerV2 replaces the grpc_log.LoggerV2 with the input zap.Logger
+// This logger adheres to the grpc go environment variables GRPC_GO_LOG_VERBOSITY_LEVEL and GRPC_GO_LOG_SEVERITY_LEVEL.
```

Comment:
What is the use case for adding the extra log severity level handling instead of accepting the log level of the zap.Logger as-is?

Reply:
If I want my application to log at the info level and want the grpc system to log at error using the same global logger instance. Plus matches https://github.com/grpc/grpc-go/blob/24b2fb8959201be9ce659bc87b0d590a34c67eae/grpclog/loggerv2.go#L124. Thoughts? I can remove the severity level if you like and we could add it in the future if we feel the need.

Reply:
I would favour starting with the explicit version: use the incoming logger's severity - and for verbosity, accept an argument with the desired verbosity level.

Reply:
What's your thoughts on having an option to allow for setting verbosity level and defaulting to 0?

Reply:
Updated @devnev